<?xml version="1.0" ?>
<sdf version="1.6">
  <model name="small_gear_tray">
    <static>false</static>
    <allow_auto_disable>false</allow_auto_disable>

    <link name="small_gear_tray">
      <inertial>
        <pose>0 0 0.02 0 0 1.57</pose>
        <mass>0.7</mass>
        <inertia>
          <ixx>0.00108</ixx>
          <ixy>0</ixy>
          <ixz>0</ixz>
          <iyy>0.00024</iyy>
          <iyz>0</iyz>
          <izz>0.00113</izz>
        </inertia>
      </inertial>
      
      <visual name="small_gear_tray">
        <geometry>
          <mesh>
            <uri>model://demo_parts/small_gear_tray/meshes/small_gear_part_tray.stl</uri>
          </mesh>
        </geometry>
        <material>
          <ambient>1.0 1.0 1.0 1.0</ambient>
          <diffuse>1.0 1.0 1.0 1.0</diffuse>
        </material>
      </visual>

      <collision name="small_gear_tray_collision">
        <pose>0 0 0 0 0 0</pose>
        <geometry>
          <mesh>
            <uri>model://demo_parts/small_gear_tray/meshes/small_gear_part_tray.stl</uri>
          </mesh>
        </geometry>
        <max_contacts>5</max_contacts>
        <surface>
          <contact>
            <ode>
              <max_vel>0.0</max_vel>
              <min_depth>0.01</min_depth>
            </ode>
          </contact>
          <friction>
            <ode>
              <mu>1000</mu>
              <mu2>1000</mu2>
            </ode>
          </friction>
        </surface>
      </collision>

      <sensor name='small_gear_tray_contact_sensor' type='contact'>
        <contact>
          <collision>small_gear_tray_collision</collision>
        </contact>
      </sensor>
    </link>
  
    <plugin name="ariac_sensors::VacuumGripperPlugin" filename="libVacuumGripperPlugin.so">
      <ros>
      </ros>
      <gripper_link>small_gear_tray_gripper</gripper_link>
      <!-- <robot_name>${prefix}_robot</robot_name> -->
      <!-- <gz_contact_topic>/world/test/model/small_gear_tray_0/link/small_gear_tray/sensor/small_gear_tray_contact_sensor/contact</gz_contact_topic> -->
    </plugin>
  </model>
</sdf>